spec/requests/servers_request_spec.rb

```diff
@@ -0,0 +1,257 @@
+  let(:server) { servers(:one) }
+  let(:server2) { servers(:two) }
+
+  before do
+    sign_in users(:one)
+
+    server.save
+    server2.save
+  end
+
+  describe "GET /index" do
+    before { get servers_path }
+
+    it { expect(response).to have_http_status(:success) }
+    it { expect(response).to render_template(:index) }
+    it { expect(response.body).to include(server.name) }
+    it { expect(response.body).to include(server2.name) }
+  end
+
+  describe "GET /show" do
+    before { get server_path(server) }
+
+    it { expect(response).to have_http_status(:success) }
+    it { expect(response).to render_template(:show) }
+    it { expect(response.body).to include(server.name) }
+
+    context "while using id" do
+      before { get server_path(server.id) }
+
+      it { expect(response).to have_http_status(:success) }
+    end
+
+    context "while using name" do
+      before { get server_path(server.name) }
+
+      it { expect(response).to render_template(:show) }
+    end
+
+    context "while using serial number" do
+      before { get server_path(server.numero) }
+
+      it { expect(response).to render_template(:show) }
+    end
+  end
+
+  describe "GET /new" do
+    before { get new_server_path }
+
+    it { expect(response).to have_http_status(:success) }
+    it { expect(response).to render_template(:new) }
+  end
+
+  describe "GET /duplicate" do
+    before { get duplicate_server_path(server) }
+
+    it { expect(response).to have_http_status(:success) }
+    it { expect(response).to render_template(:duplicate) }
+  end
+
+  describe "POST /create" do
+    context "with valid parameters" do
+      let(:valid_attributes) do
+        server.attributes.except(["id", "numero", "name"]).merge(numero: "new_numero", name: "NewServerName")
+      end
+
+      it "creates a new Server" do
+        expect do
+          post servers_path, params: { server: valid_attributes }
+        end.to change(Server, :count).by(1)
+      end
+
+      it "redirects to the created server" do
+        post servers_path, params: { server: valid_attributes }
+        expect(response).to redirect_to(assigns(:server))
+      end
+    end
+
+    context "with invalid parameters" do
+      let(:invalid_attributes) { { name: "" } }
+
+      it "does not create a new Server without attributes" do
+        expect { post servers_path, params: { server: {} } }
+          .to raise_error(ActionController::ParameterMissing)
+      end
+
+      it "does not create a new Server without parameters" do
+        expect { post servers_path, params: {} }
+          .to raise_error(ActionController::ParameterMissing)
+      end
+
+      it "does not create a new Server with an existing serial number" do
+        expect do
+          post servers_path, params: { server: server2.attributes.except(["id"]) }
+        end.not_to change(Server, :count)
+      end
+
+      it "does not create a new Server with invalid parameters", :aggregate_failures do
```

Comment:
```suggestion
      it "does not create a new Server with invalid parameters" do
```
It seems not to be necessary here